src/btree/bt_sync.c

```diff
@@ -100,6 +100,88 @@ __sync_dup_walk(WT_SESSION_IMPL *session, WT_REF *walk, uint32_t flags, WT_REF *
     return (0);
 }
 
+/*
+ * __wt_ref_is_obsolete --
+ *     Check whether the ref is obsolete according to the newest stop time pair.
+ */
+static bool
+__wt_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+    WT_ADDR *addr;
+    bool obsolete;
+
+    obsolete = false;
+
+    addr = ref->addr;
+    if (addr && __wt_txn_visible_all(session, addr->newest_stop_txn, addr->newest_stop_ts))
+        obsolete = true;
+
+    return obsolete;
```

Comment:
I find this code clearer without the `obsolete` boolean flag.  E.g.
```
  return (addr && __wt_txn_visible_all(session, addr->newest_stop_txn, addr->newest_stop_ts));
```

Reply:
This function adds some more additional conditions later to handle in-memory pages, so I left it as it is currently.